client/client_dns_test.go

```diff
@@ -25,14 +25,12 @@ func TestResolveKDC(t *testing.T) {
 	if err != nil {
 		t.Errorf("error resolving KDC via DNS TCP: %v", err)
 	}
-	assert.Equal(t, 5, count, "Number of SRV records not as expected: %v", res)
+	// This should be 1 because for the default realm of TEST.GOKRB5,
+	// only 1 kdc is set.
+	assert.Equal(t, 1, count, "Number of SRV records not as expected: %v", res)
 	assert.Equal(t, count, len(res), "Map size does not match: %v", res)
 	expected := []string{
-		"kdc.test.gokrb5:88",
-		"kdc1a.test.gokrb5:88",
-		"kdc2a.test.gokrb5:88",
-		"kdc1b.test.gokrb5:88",
-		"kdc2b.test.gokrb5:88",
+		"127.0.0.1:88",
```

Comment:
Hi, I think the original test is still needed but with a configuration that does not include a kdc defined in the realms section. Then this test you have here is also needed to cover the fact that a specifically defined kdc should override the dns_lookup_kdc setting.

Therefore I propose keeping the TestResolveKDC, rename to TestResolveKDC_DNS and just add this line:
```
c.Realms = make([]config.Realm, 0)
```

Then also create new new test called TestResolveKDC_OverrideDNS and make it's content what you have here other than you should not need the line:
```
test.Privileged(t)
```

FYI:
The ``test.Privileged(t)`` skips the test if it is not executed as an integration test with high privileges in the environment in which it is run. In this case those privileges are used to set the resolv.conf to look at a DNS server that has been created for the integration test. For more on this take a look at the set up of integration test dependencies in this step of the github actions:
https://github.com/jcmturner/gokrb5/blob/039b60d3e315310b951fcad322377f96f0924969/.github/workflows/testing.yml#L49-L62

Reply:
As I've continued my review I see you already have config/hosts_test.go/TestConfig_GetKDCsUsesConfiguredKDC which does the equivalent of the TestResolveKDC_OverrideDNS I am proposing here. Therefore I think we just need to revert this integration test with the minor modification.